### PR TITLE
Fluffy: Implement poke in state network

### PR DESCRIPTION
**Pull request by bhartnett**
Poke in the state network is feasible by passing in the parent offer when looking up content. The fetched state content is then combined with the parent offer to build the offer which is used in the poke. 

I'm going to suggest that poke is added to the state network specs because I can't see any reason for it to be disabled after implementing this change.

I've also added poke to the stateGetContent JSON-RPC method by supporting an optional parentContent parameter which should contain the parent offer which is required in order to enable poke.

### CODE REVIEW

**Comment by bhartnett** [comment]
Leaving this as a draft PR until the specs are updated

File: fluffy/rpc/rpc_portal_state_api.nim [hunk 1]
```diff
@@ -83,7 +83,9 @@ proc installPortalStateApiHandlers*(rpcServer: RpcServer, p: PortalProtocol) =
 
     SSZ.encode(offerResult).to0xHex()
 
-  rpcServer.rpc("portal_stateGetContent") do(contentKey: string) -> ContentInfo:
+  rpcServer.rpc("portal_stateGetContent") do(
+    contentKey: string, parentContent: Opt[string]
+  ) -> ContentInfo:
```
Comment (kdeme) [inline fluffy/rpc/rpc_portal_state_api.nim:88]: For a user this API does complicate things. Feels like the "health" of the network should not be pushed towards the UX of user API. But perhaps its ok as an `Opt`...

**Review by kdeme — COMMENTED** [review]
lgtm but would wait with merging until accepted in specs first.

File: fluffy/rpc/rpc_portal_state_api.nim (hunk 1 above)
Comment (bhartnett) [reply fluffy/rpc/rpc_portal_state_api.nim:88]: Actually I agree and for this reason I didn't propose changing the JSON-RPC spec to add this in my other PR. I'm planning to revert this part but I'll wait to see what happens with the spec changes first.

**Pull request merged**